app/jobs/link_discourse_account_job.rb

```diff
@@ -13,31 +13,39 @@ def perform(user)
     discourse_user = discourse.find_user_by_email
 
     if discourse_user
-      if user.confirmed?
-        user.update(external_id: discourse_user["id"])
-      else
-        # send confirmation email
-        User.send_confirmation_instructions(email: user.email)
-      end
+      user.external_id = discourse_user["id"]
+      user.username = discourse_user["username"]
 
-      return
-    end
+      response = discourse.create_email_token
 
-    # Create discourse account
-    response = discourse.create_user
+      if [true, "OK"].exclude?(response["success"])
+        return Raven.capture_message(
+          "Couldn't create a Discourse invite",
+          extra: {user_id: user.id, user_email: user.email}
+        )
+      end
+
+      user.email_token = response["email_token"]
+      logger.info("Discourse account #{discourse_user["id"]} linked with user id #{user.id}")
+    else
+      # Create discourse account
+      response = discourse.create_user
+
+      if [true, "OK"].exclude?(response["success"])
+        return Raven.capture_message(
+          "Couldn't create a Discourse invite",
+          extra: {user_id: user.id, user_email: user.email}
+        )
+      end
 
-    if [true, "OK"].exclude?(response["success"])
-      return Raven.capture_message(
-        "Couldn't create a Discourse invite",
-        extra: {user_id: user.id, user_email: user.email}
-      )
+      user.email_token = response["email_token"]
+      user.external_id = response["user_id"]
+      user.username = response["username"]
+      logger.info("Discourse account created. discourse_id: #{response["user_id"]} user_id: #{user.id}")
```

Comment:
This is ok to be here?
_I am cool with it as it will offer a better DX, but just making sure this is not a sort of `console.log` unintended_

Reply:
Yes, I'm trying to add logging for the most important flows of the app.